### PR TITLE
docs(docs-infra): add top level banner component

### PR DESCRIPTION
**Pull request by 9kubczas4**
docs(docs-infra): add top level banner component

- create top level banner component
- write unit tests
- close banner and keep state in the local storage

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x ] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x ] No

### CODE REVIEW

File: adev/src/app/app.component.html [hunk 1]
```diff
@@ -1,5 +1,6 @@
 @defer (when isBrowser) {
   <adev-progress-bar />
+  <docs-top-level-banner id="ng-survey-2024" link="https://goo.gle/angular-v18" text="Take the Angular Developer Survey today!"  />
```
Comment (JeanMeche) [inline adev/src/app/app.component.html:3]: Comment that (once i

**Review by JeanMeche — APPROVED** [review]

**Comment by shannonvc** [comment]
I am not sure if this is a use case that was expected to work but on small devices the new component appears to be hidden behind the mobile nav bar. 
image

It's only visible when the navigation is opened
image

**Comment by 9kubczas4** [comment]
Thank you @shannonvc, I've started working on support for smaller screens.

Still I need to solve following issues:
- tablets: display banner below navbar (pages with subnavigation), banner is behind secondary nav 
image
- mobile: I moved the banner below the navbar like below 
image
but it should be moved to the top or hidden when sidenav was opened.

**Comment by 9kubczas4** [comment]
Fixes of the issues related to screen responsiveness were applied in the recent commit.

**Review by twerske — APPROVED** [review]
LGTM

Can you comment the banner out for now and we'll approve. I'll PR a few CSS style changes to the animation on the banner in a follow up PR based on deign feedback we've gotten!

Thank you @9kubczas4 !!!!!

**Pull request closed without merging**